browserdebuggertools/wssessionmanager.py

```diff
@@ -15,23 +17,116 @@
 from browserdebuggertools.exceptions import (
     DevToolsException, ResultNotFoundError, TabNotFoundError, MaxRetriesException,
     DevToolsTimeoutException, DomainNotEnabledError,
-    MethodNotFoundError, UnknownError, ResourceNotFoundError, TimerException
+    MethodNotFoundError, UnknownError, ResourceNotFoundError, DeadMessagingThread
 )
 
 
-def open_connection_if_closed(socket_handler_method):
+class _WSMessagingThread(Thread):
+
+    _CONN_TIMEOUT = 15
+    _BLOCKED_TIMEOUT = 5
+    _MAX_QUEUE_BUFFER = 1000
+    _POLL_INTERVAL = 1
+
+    def __init__(self, port, send_queue, recv_queue, poll_signal):
```

Comment:
I don't think `_WSMessagingThread` needs to be passed the send and recv queues. They should be private.
Further, you would probably want `collections.deque` here instead of vanilla lists. They are more efficient for the queue operations you are interested in, e.g. `pop(0)` and have the additional advantage of being thread safe (to be fair, the interaction with the session manager looks thread safe to me, but the code my change in the future and we may not notice it is not thread-safe anymore).

Reply:
Hey @RodolfoSGomez I think we do want WSSessionManager to hold a reference to the queues and pass them to the messaging thread. That way if we have to throw the messaging thread away because we don't lose what's in the queues. I'll look into collections.deque, thanks.

Reply:
Good point